### DISCUSSION TITLE
How to add (and persist) menu entries to GRUB menu?

### DISCUSSION THREAD

**Discussion by C-Duv** [minios-linux/minios-live#53] [Q&A] [answered]
I want to setup multiboot on my computer using MiniOS as a base (and it's GRUB) with other OS (such as LibreELEC).

I've booted MiniOS from an USB then installed it on the internal hard drive (using the provided tool).
With GParted I have created the partition and I know the GRUB `menuentry` to add for LibreELEC but I can't make my changes persist.

Changes to both files `/etc/minios/grub.cfg` and `/run/initramfs/memory/changes/etc/minios/grub.cfg` are being reverted on shutdown.

What's the correct way to do so?

I could not find any documentation on how to "update GRUB" on MiniOS.

**Comment by C-Duv** [comment]
OK, it looks like the correct file to update is `/run/initramfs/memory/data/minios/boot/grub/grub.cfg`, after a reboot it's content gets "propagated" to `/run/initramfs/memory/changes/etc/minios/grub.cfg` (and thus to `/etc/minios/grub.cfg` also).

Note: It needs to be made _writable_ first: `chmod u+w /run/initramfs/memory/data/minios/boot/grub/grub.cfg`.

**Reply by crim50n** [reply]
You only need to edit the files in /etc/minios, they will automatically replace the original files on the flash drive when you shut down the system.

**Reply by C-Duv** [reply]
I'll double-check but I'm pretty sure it did not worked.

**Reply by crim50n** [reply]
Okay, I'll check it out.

**Comment by crim50n** [comment, marked as answer]
Yes, indeed, changes must be made to the configuration file on the flash drive. The file in /etc/minios is for rebuilding the ISO.
